src/table/settings/gui_settings.ini

```diff
@@ -130,12 +130,17 @@ def      = false
 str      = STR_CONFIG_SETTING_SMOOTH_SCROLLING
 strhelp  = STR_CONFIG_SETTING_SMOOTH_SCROLLING_HELPTEXT
 
-[SDTC_BOOL]
-var      = gui.right_mouse_wnd_close
```

Comment:
I do not like that we reset the setting of everything. As a player, that would be rather frustrating: because someone wanted to add an option, everyone has to suffer, basically ;)

But, it is not the first time we encounter this situation .. and the solution takes a tiny bit of work.

The best example I know of, is `_old_units`. You can grep the code for that, and see how it is done. It does require a savegame bump to be handled properly, which might be a bit complicated for your first PR, so if you can't figure it out, do ask for help. But basically:

Keep the old setting around, and use `var` to store it in an `_old_NNN` variable. Add a new `SLV_NNN` to bump the savegame version. In `afterload.cpp`, check all savegames before the new version, and use the `_old_NNN` to write to the new `right_click_wnd_close` to correct value.

This way, nobody that was using "yes" will notice any difference.

PS: `NNN` ofcourse is some name you have to come up with. Not literally `NNN` ;)

Reply:
Okay, forget that, there is an easier way to do this. When you changed the type to `OMANY`, we can simply define a `load` to do conversion. Besides, my earlier story assumed we saved this setting in the savegame .. we don't. Even easier :D

Basically, with `OMANY`, it first tries to use the supplied list to convert the string to a number. If that fails, it will check if a `load` function is defined. Here you can convert `true` / `false` to the correct value. An example of this is `ConvertLandscape`, and a very recent PR I made to confirm this actually works, adding `ConvertAutosave`:

https://github.com/OpenTTD/OpenTTD/pull/11143

As you can see in that PR, it didn't work out-of-the-box, as the `load` was never connected in the `SDTC_OMANY` yet. But that is fixed wit that PR :)

If you follow that, the code becomes much easier. You do need to rename the variable back to `right_mouse_wnd_close` for it to work. But you do need to wait for that PR to be merged, ofc :)